### PR TITLE
[Feature] Support ilike function

### LINKED ISSUE #14595: [feature] need ilike
SR has the function `like` but it is only case-sensitive function.
To support more scenario, we need another function named `ilike`, which is similar to `like` but is case-insensitive.

To be more specific:

1. what we have now:
```
mysql> select like("star","star");
+----------------------+
| like('star', 'star') |
+----------------------+
|                    1 |
+----------------------+

MySQL [(none)]> select ilike('star','STAr');
+----------------------+
| like('star', 'STAr') |
+----------------------+
|                    0 |
+----------------------+
1 row in set (0.00 sec)
```

what the ilike will perform:
```
mysql> select ilike("star","star");
+----------------------+
| like('star', 'star') |
+----------------------+
|                    1 |
+----------------------+

MySQL [(none)]> select ilike('star','STAr');
+----------------------+
| like('star', 'STAr') |
+----------------------+
|                    1 |
+----------------------+
1 row in set (0.00 sec)
```

### PR DESCRIPTION
**Pull request by uncleGen**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #14595

## Problem Summary(Required) ：

SR has the function like but it is only case-sensitive function.
To support more scenario, we need another function named ilike, which is similar to like but is case-insensitive.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java [hunk 1]
```diff
@@ -343,4 +348,15 @@ private static ScalarOperator ifCall(CallOperator call) {
 
         return ((ConstantOperator) call.getChild(0)).getBoolean() ? call.getChild(1) : call.getChild(2);
     }
+
+    private static ScalarOperator ilikeCall(CallOperator call) {
```
Comment (uncleGen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java:352]: convert `ilike` to `like` by lowerring string

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunction.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java|10|10|100.00%|[]|

**Review by stdpain — DISMISSED** [review]

**Comment by stdpain** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunction.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java|10|10|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 12 / 12 (100.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/TableFunction.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java|10|10|100.00%|[]|

**Comment by uncleGen** [comment]
ping

**Review by stdpain — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunction.java [hunk 2]
```diff
@@ -82,6 +83,9 @@ public static void initBuiltins(FunctionSet functionSet) {
                 new TableFunction(new FunctionName("json_each"), Lists.newArrayList("key", "value"),
                         Lists.newArrayList(Type.JSON), Lists.newArrayList(Type.VARCHAR, Type.JSON));
         functionSet.addBuiltin(jsonEachFunction);
+
+        functionSet.addBuiltin(ScalarFunction.createBuiltinOperator(
+                FunctionSet.ILIKE, Strings.EMPTY, Lists.newArrayList(Type.VARCHAR, Type.VARCHAR), Type.BOOLEAN));
```
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunction.java:88]: register function in `functions.py` plz

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java (hunk 1 above)
Comment (Seaven) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java:352]: could you move the function to `NormalizePredicateRule` ？

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunction.java (hunk 2 above)
Comment (uncleGen) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableFunction.java:88]: ok

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java (hunk 1 above)
Comment (uncleGen) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/SimplifiedPredicateRule.java:352]: ok

**Comment by uncleGen** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Review by stdpain — APPROVED** [review]

**Comment by Seaven** [comment]
@evelynzhaojie cc, add function

**Pull request merged**